modules/cudaoptflow/src/nvidiaOpticalFlow.cpp

```diff
@@ -0,0 +1,548 @@
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+#include "precomp.hpp"
+
+#if !defined HAVE_CUDA || defined(CUDA_DISABLER)
+
+Ptr<NvidiaOpticalFlow_1_0> cv::cuda::NvidiaOpticalFlow_1_0::create(int, int, int, NV_OF_PERF_LEVEL, NV_OF_BOOL, NV_OF_BOOL) { throw_no_cuda(); return Ptr<NvidiaOpticalFlow_1_0>(); }
+
+#else
+
+#if defined(_WIN32) || defined(_WIN64)
+#include <Windows.h>
+#else
+#define HMODULE void *
+#define _stricmp strcasecmp
+#include <dlfcn.h>
+#endif
+
+//macro for dll loading
+#if defined(_WIN64)
+#define MODULENAME TEXT("nvofapi64.dll")
+#elif defined(_WIN32)
+#define MODULENAME TEXT("nvofapi.dll")
+#else
+#define MODULENAME "libnvidia-opticalflow.so.1"
+#endif
+
+#define NVOF_API_CALL(nvOFAPI)                                                      \
+    do                                                                              \
+    {                                                                               \
+        NV_OF_STATUS errorCode = nvOFAPI;                                           \
+        if (errorCode != NV_OF_SUCCESS)                                             \
+        {                                                                           \
+            std::ostringstream errorLog;                                            \
+            errorLog << #nvOFAPI << "returned error " << (unsigned int)errorCode;   \
+            CV_Error(Error::StsError, errorLog.str());                              \
+        }                                                                           \
+    } while (0)
+
+using namespace std;
+using namespace cv;
+using namespace cv::cuda;
+
+namespace
+{
+    class NvidiaOpticalFlowImpl : public cv::cuda::NvidiaOpticalFlow_1_0
```

Comment:
Please remove indentation in namespaces.

Reply:
Sure. Noted